engine/client_internal.go

```diff
@@ -155,10 +156,46 @@ func (c *Client) loadPaymailComponents() (err error) {
 	return
 }
 
+func (c *Client) loadPaymailAddressService() error {
+	if c.options.paymailAddressService == nil {
```

Comment:
Maybe explicit return nil error if the service is already created? ...and skip the wrapping? 🤔 
```
if c.options.paymailAddressService != nil {
    return nil
```
//...rest of your code
return nil